### PR TITLE
[Component with its own class name as service type ] LPS-199482 Refactor classes JAXRSLifecycle and KaleoTaskScriptedAssignmentCache

### PR DESCRIPTION
**Pull request by Valmir-unicap**
- Ticket: https://liferay.atlassian.net/browse/LPS-199482

- Correction of PR from @RenanUmV: https://github.com/rafaprax/liferay-portal/pull/1347

### CODE REVIEW

**Review by rafaprax — CHANGES_REQUESTED** [review]

File: modules/apps/portal-remote/portal-remote-jaxrs-whiteboard/src/main/java/com/liferay/portal/remote/jaxrs/whiteboard/lifecycle/JAXRSLifecycle.java [hunk 1]
```diff
@@ -21,12 +21,6 @@
 public class JAXRSLifecycle {
 
 	public void ensureReady() {
-		if (_jaxrsReady) {
```
[the remark this answers is not in the document]
Comment (Valmir-unicap) [reply modules/apps/portal-remote/portal-remote-jaxrs-whiteboard/src/main/java/com/liferay/portal/remote/jaxrs/whiteboard/lifecycle/JAXRSLifecycle.java:24]: I did what you suggested: 

- @RenanUmV, first remove the _jaxrsReady variable and then transform it to a Util class. Also, verify a solution for the _serviceRegistrationDCLSingleton.destroy call as well.

[the remark this answers is not in the document]
Comment (rafaprax) [reply modules/apps/portal-remote/portal-remote-jaxrs-whiteboard/src/main/java/com/liferay/portal/remote/jaxrs/whiteboard/lifecycle/JAXRSLifecycle.java:24]: I know but you need to explain it in the commit message. I'm just the first reviewer in the chain and your current commit message doesn't explain the reason why you did that.

[the remark this answers is not in the document]
Comment (rafaprax) [reply modules/apps/portal-remote/portal-remote-jaxrs-whiteboard/src/main/java/com/liferay/portal/remote/jaxrs/whiteboard/lifecycle/JAXRSLifecycle.java:24]: Makes sense?

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
ci:test:sf

File: modules/apps/portal-remote/portal-remote-jaxrs-whiteboard/src/main/java/com/liferay/portal/remote/jaxrs/whiteboard/util/JAXRSLifecycleUtil.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		_serviceRegistrationDCLSingleton.getSingleton(
+			() -> bundle.getBundleContext(
+			).registerService(
+				Object.class, new Object(),
+				MapUtil.singletonDictionary(
+					"liferay.jaxrs.whiteboard.ready", true)
+			));
+	}
+
+	public static void ensureUnReady() {
```
Comment (rafaprax) [inline modules/apps/portal-remote/portal-remote-jaxrs-whiteboard/src/main/java/com/liferay/portal/remote/jaxrs/whiteboard/util/JAXRSLifecycleUtil.java:32]: @Valmir-unicap rename to `ensureUnready`

**Review by rafaprax — CHANGES_REQUESTED** [review]

File: modules/apps/oauth2-provider/oauth2-provider-scope-impl/src/main/java/com/liferay/oauth2/provider/scope/internal/liferay/ScopeLocatorImpl.java [hunk 3]
```diff
@@ -54,7 +54,7 @@ public class ScopeLocatorImpl implements ScopeLocator {
 	public LiferayOAuth2Scope getLiferayOAuth2Scope(
 		long companyId, String applicationName, String scope) {
 
-		_jaxrsLifecycle.ensureReady();
+		JAXRSLifecycleUtil.ensureReady();
```
Comment (rafaprax) [inline modules/apps/oauth2-provider/oauth2-provider-scope-impl/src/main/java/com/liferay/oauth2/provider/scope/internal/liferay/ScopeLocatorImpl.java:57]: Hey @Valmir-unicap, please add a breaking change note for this commit. I didn't see this before 😞

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Review by rafaprax — CHANGES_REQUESTED** [review]
@Valmir-unicap the breaking change you made wasn't in the `bnd.bnd` file. This `bnd.bnd`'s change is a result of the breaking change. https://github.com/rafaprax/liferay-portal/pull/1372/commits/379cd814489393e971b13773f3c4abf48d4654c0

Please fix the commit message to reflect that and pay attention to the details. Ask @MrJohn1911 help and review for that.

Thx

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Review by rafaprax — APPROVED** [review]

**Comment by rafaprax** [comment]
forwarded to https://github.com/liferay-core-infra/liferay-portal/pull/6363

**Pull request closed without merging**